SwiftExample/SwiftExample/ChatViewController.swift

```diff
@@ -55,11 +58,110 @@ class ChatViewController: JSQMessagesViewController {
         self.messages = makeConversation()
         self.collectionView?.reloadData()
         self.collectionView?.layoutIfNeeded()
+    }
+    
+    func receiveMessagePressed(sender: UIBarButtonItem) {
+        /**
+         *  DEMO ONLY
+         *
+         *  The following is simply to simulate received messages for the demo.
+         *  Do not actually do this.
+         */
         
         
+        /**
+         *  Show the typing indicator to be shown
+         */
+        self.showTypingIndicator = !self.showTypingIndicator
+        
+        /**
+         *  Scroll to actually view the indicator
+         */
+        self.scrollToBottomAnimated(true)
+        
+        /**
+         *  Copy last sent message, this will be the new "received" message
+         */
+        var copyMessage = self.messages.last?.copy()
+        
+        if (copyMessage == nil) {
+            copyMessage = JSQMessage(senderId: AvatarIdJobs, displayName: DisplayNameJobs, text: "First received!")
+        }
+            
+        var newMessage:JSQMessage?
+        var newMediaData:JSQMessageMediaData?
+        var newMediaAttachmentCopy:AnyObject?
+        
+        if copyMessage!.isMediaMessage() {
+            /**
+             *  Last message was a media message
+             */
+            let copyMediaData = copyMessage!.media
+            
+            switch copyMediaData {
+            case is JSQPhotoMediaItem:
+                let photoItemCopy = (copyMediaData as! JSQPhotoMediaItem).copy() as! JSQPhotoMediaItem
+                photoItemCopy.appliesMediaViewMaskAsOutgoing = false
+                
+                newMediaAttachmentCopy = UIImage(CGImage: photoItemCopy.image.CGImage!)
+                
+                /**
+                 *  Set image to nil to simulate "downloading" the image
+                 *  and show the placeholder view5017
+                 */
+                photoItemCopy.image = nil;
+                
+                newMediaData = photoItemCopy
+            default:
+                print("error: unrecognized media item")
```

Comment:
We Should not make this a print statement that will be ignored. Lets go ahead and do an 
`assertionFailure("Error: This Media type was not recognised")`